### PR TITLE
Bug 1218303 - Check history records by time and export maximum size argument to SyncBrowserDB.getHistory. r=yifan

### PR DESCRIPTION
**Pull request by weilonge**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=05e77ac3eaaeb870d42ef145267b3535aeeb4c8a

File: tv_apps/browser/js/sync/db.js [hunk 1]
```diff
@@ -1394,6 +1396,32 @@ SyncBrowserDB.db = {
   },
 
   /**
+   * Check history by timestamp range at (record.timestamp <= start)
+   * @param {Number} start Start timestamp
+   * @param {Function} callback Runs on complete with an array of history
```
Comment (begeeben) [inline tv_apps/browser/js/sync/db.js]: I'm confused about the  mismatch of the value passed in the callback and the comment.

File: tv_apps/browser/js/sync/db.js [hunk 2]
```diff
@@ -1394,6 +1396,32 @@ SyncBrowserDB.db = {
...
+   */
+  checkHistoryByTime: function db_checkHistoryByTime(start, callback) {
```
Comment (begeeben) [inline tv_apps/browser/js/sync/db.js]: ```getPreviousHistoryTimestamp``` sounds more fit to the usage of this function. What do you think?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ebdd93b302395f0b6b12cb69e04fcb6b243ad490

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b24df87830eb348ff54f6d7202d9b681609757bc

**Pull request merged**